src/daemon/plugin.c

```diff
@@ -1779,14 +1772,10 @@ EXPORT int plugin_read_all_once(void) {
     old_ctx = plugin_set_ctx(rf->rf_ctx);
 
     if (rf->rf_type == RF_SIMPLE) {
-      int (*callback)(void);
-
-      callback = rf->rf_callback;
+      int (*callback)(void) = (void *)rf->rf_callback;
```

Comment:
Shouldn't the cast be for the exact type instead of `(void *)` ?

Reply:
Discussed above.